hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/ConfigurationProperties.java

```diff
@@ -158,14 +190,24 @@ private void copyProperties(
    */
   private void storePropertiesInPrefixNodes(Map<String, String> props) {
     for (Map.Entry<String, String> prop : props.entrySet()) {
-      List<String> propertyKeyParts = splitPropertyByDelimiter(prop.getKey());
-      if (!propertyKeyParts.isEmpty()) {
-        PrefixNode node = findOrCreatePrefixNode(nodes,
-            propertyKeyParts.iterator());
-        node.getValues().put(prop.getKey(), prop.getValue());
-      } else {
-        LOG.warn("Empty configuration property, skipping...");
-      }
+      getNode(prop.getKey()).ifPresent(
+          node -> node.getValues().put(prop.getKey(), prop.getValue()));
+    }
+  }
+
+  /**
+   * Finds the node that matches the whole key or create it, if it does not exist.
+   * @param name name of the property
+   * @return the found or created node, if the name is empty, than empty optional will return
+   */
+  private Optional<PrefixNode> getNode(String name) {
```

Comment:
Optional is only ergonomic when used in tandem with the Streaming API. I think we are better off with the classic solution here (explicit nulls).